src/base/static/client/mapseed-api-client.js

```diff
@@ -58,115 +58,339 @@ const deletePlaceTag = async url => {
+      placePagePromise.then(featureCollection =>
+        fromGeoJSONFeatureCollection({
+          featureCollection,
+          datasetSlug,
+          clientSlug,
+        }),
+      ),
+    );
+  } catch (err) {
+    // eslint-disable-next-line no-console
+    console.error("Error: Failed to fetch places.", err);
+  }
+};
 
-        success: function(fetchedCollection) {
-          resolve(fetchedCollection, collectionId);
+// TODO: include_private_places and include_private_fields ??
+const updatePlace = async (placeUrl, placeData) => {
+  try {
+    placeData = toGeoJSONFeature(placeData);
+    return await fetch(
+      `${placeUrl}${buildQuerystring({
+        include_tags: true,
+        include_submissions: true,
+      })}`,
+      {
+        headers: {
+          "Content-Type": "application/json",
+          "X-Shareabouts-Silent": true, // To prevent new Actions on update.
         },
+        credentials: "include",
+        method: "PUT",
+        body: JSON.stringify(placeData),
+      },
+    )
+      .then(status)
+      .then(json);
+  } catch (err) {
+    // eslint-disable-next-line no-console
+    console.error("Error: Failed to update Place.", err);
+  }
+};
 
-        error: function(collection, err) {
-          layer.is_visible_default && setLayerError(collectionId);
-          // eslint-disable-next-line no-console
-          console.error(
-            `error loading place collection: ${collectionId}: err:`,
-            err,
-          );
-          reject(err);
-          // TODO: layer loading event; fix in layer UI PR
-        },
+const createPlace = async ({
+  datasetUrl,
+  placeData,
+  datasetSlug,
+  clientSlug,
+}) => {
+  try {
+    placeData = toGeoJSONFeature(placeData);
+    return await fetch(`${datasetUrl}/places?include_tags`, {
+      headers: {
+        "Content-Type": "application/json",
+      },
+      credentials: "include",
+      method: "POST",
+      body: JSON.stringify(placeData),
+    })
+      .then(status)
+      .then(json)
+      .then(feature =>
+        fromGeoJSONFeature({ feature, datasetSlug, clientSlug }),
+      );
+  } catch (err) {
+    // eslint-disable-next-line no-console
+    console.error("Error: Failed to create Place.", err);
+  }
+};
+
+const getActivity = async (datasetUrl, clientSlug) => {
+  try {
+    return await fetch(`${datasetUrl}/actions`, {
+      credentials: "include",
+    })
+      .then(status)
+      .then(json)
+      .then(response => {
+        return Promise.resolve(
+          // Note that activity is returned from the API in paginated form.
+          // Here we load just the first page, assuming only the most
+          // recent page of activity is useful to the user.
+          response.results.map(activity => ({
+            ...activity,
+            // Add the "client slug" to be used with activity from this dataset.
+            // The client slug is used in urls in the client.
+            _clientSlug: clientSlug,
```

Comment:
I introduced the idea of a "client slug," which is the term for `idea` in a url like `/idea/1234`. I think in the past we discussed removing this slug entirely (or forcing it to always be a generic term like `place`), but for now we keep the client slug.